lib/translator/read.cc

```diff
@@ -0,0 +1,187 @@
+#include "read.h"
+
+#include "absl/types/span.h"
+
+#include "common.h"
+
+namespace translator {
+
+namespace {  // anonymous namespace for helper functions
+
+// Section 5.3
+// https://www.nvmexpress.org/wp-content/uploads/NVM-Express-SCSI-Translation-Reference-1_1-Gold.pdf
+StatusCode BuildPrinfo(uint8_t rdprotect, uint8_t& prinfo) {
+  bool pract;     // Protection Information Action 1 bit
+  uint8_t prchk;  // Protection Information Check 3 bits
+  prinfo = 0;     // Protection Information field 4 bits
```

Comment:
Small nit, but this will write to the out-parameter even if the following switch statement returns an error code.  In general, having out-parameters be undefined in the case of errors is ok, but it's something to consider.

Reply:
Instead of setting this to zero and or-ing prchk, you could just assign prchk below.